etils/etree/backend.py

```diff
@@ -116,6 +116,15 @@ def assert_same_structure(
       )
 
 
+class Optree(Jax):
+  """`optree` backend."""
```

Comment:
FYI, `jax.tree_util` always treat `None` as internal node in `PyTreeDef`. But `tf.nest` and `dm-tree` will flatten `None` into the leaves list. This means switching pytree backend will give different results. `optree` supports both cases and provides an extra argument `none_is_leaf` (defaults to same behavior as `jax`). See [`None` is Non-Leaf Node vs. `None` is Leaf](https://github.com/metaopt/optree#none-is-non-leaf-node-vs-none-is-leaf) for more details.

Reply:
Thanks for the clarification. That's an important gotcha. For now, keeping the `None` behaviour is good I think